File: MASS-FE/src/features/auth/services/auth.service.js
import axiosClient from "../../../shared/services/axiosClient";

const authService = {
  login: (payload) => {
    return axiosClient.post("/api/v1/auths/login", payload);
  },
  register: (payload) => {
    return axiosClient.post("/api/v1/auths/register", payload);
  },
  me: () => {
    return axiosClient.get("/api/v1/auths/me");
  },
  logout: () => {
    return axiosClient.post("/api/v1/auths/logout");
  },
  forgotPassword: (email) => {
    return axiosClient.post("/api/v1/auths/forgot-password", { email });
  },
  verifyOtp: async (email, otp) => {
    return axiosClient.post('/api/v1/auths/verify-otp', { email, otp });
  },

  verifyRegistrationOtp: async (email, otp) => {
    return axiosClient.post('/api/v1/auths/verify-registration-otp', { email, otp });
  },
  resetPassword: (email, otp, newPassword) => {
    return axiosClient.post("/api/v1/auths/reset-password", { email, otp, newPassword });
  }
};

export default authService;

File: MASS-BE/src/main/java/com/group_project/MASS/service/AuthService.java
package com.group_project.MASS.service;

import com.group_project.MASS.dto.request.LoginRequest;
import com.group_project.MASS.dto.response.AuthResponse;
import com.group_project.MASS.dto.request.RegisterRequest;
import com.group_project.MASS.dto.request.RegisterRequest;

public interface AuthService {
    String login(LoginRequest loginRequest);
    void register(RegisterRequest registerRequest);
    AuthResponse getMe(String email);
    void generateAndSendOtp(String email);
    void verifyOtp(String email, String otp);
    void resetPassword(String email, String otp, String newPassword);
}
